discord/ui/view.py

```diff
@@ -327,6 +329,23 @@ async def on_timeout(self) -> None:
         """
         pass
 
+    async def on_check_failure(self, interaction: Interaction) -> None:
+        """|coro|
+
+        A callback that is called when a :meth:`View.interaction_check` returns ``False``.
+
+        This is good to overwrite when you want a custom error message
+
+        Per default this calls :meth:`InteractionResponse.defer` if the interaction has not been responded to before
+
+        Parameters
+        -----------
+        interaction: :class:`~discord.Interaction`
+            The interaction that occurred.
+        """
+        if not interaction.response._responded:
```

Comment:
Probably better to use the public method.
```suggestion
        if not interaction.response.is_done():
```